ExchangeOnline/Export-EXOMailboxSizes.ps1

```diff
@@ -279,28 +319,50 @@ Write-Verbose "There are $mailboxCount mailboxes"
 
 if ($mailboxCount -eq 0)
 {
-    return 'There are no mailboxes found using the supplied filters'
+    throw 'There are no mailboxes found using the supplied filters'
 }
 
 #  Loop through the list of mailboxes and output the results to the CSV
 Write-Verbose 'Beginning loop through all mailboxes'
 foreach ($mailbox in $mailboxes)
 {
-    if (!$PSCmdlet.MyInvocation.BoundParameters['Verbose'].IsPresent)
+    # if InputCSV is specified, match against mailbox list
+    if ($InputCSV)
     {
-        Write-Progress -Id 1 -Activity 'EXO Mailbox Size Report' -Status "Processing $($i) of $($mailboxCount) Mailboxes --- $($mailbox.UserPrincipalName)" -PercentComplete (($i * 100) / $mailboxCount)
+        if ($mailbox.UserPrincipalName -in $csv.UserPrincipalName)
```

Comment:
We should check the csv column against the email addresses here and not against the UPN column, this means it will catch a mailbox if the email exists in against any mailbox